src/utils/hoc.test.jsx

```diff
@@ -1,52 +1,42 @@
 import React from 'react';
-import { mount } from 'enzyme';
-
+import {
+  render, screen, fireEvent, waitFor,
+} from '@testing-library/react';
 import { withLocation, withNavigate, withParams } from './hoc';
 
 const mockedNavigator = jest.fn();
 
 jest.mock('react-router-dom', () => ({
   useNavigate: () => mockedNavigator,
-  useLocation: () => ({
-    pathname: '/current-location',
-  }),
-  useParams: () => ({
-    someParameter: 'some-parameter',
-  }),
+  useLocation: () => ({ pathname: '/current-location' }),
+  useParams: () => ({ someParameter: 'some-parameter' }),
 }));
 
 // eslint-disable-next-line react/prop-types
 const MockComponent = ({ navigate, location, someParameter }) => (
   // eslint-disable-next-line react/button-has-type
-  <button id="btn" onClick={() => navigate('/some-route')}>
+  <button data-testid="btn" onClick={() => navigate('/some-route')}>
     {/* eslint-disable-next-line react/prop-types */}
     {`${location.pathname}/${someParameter}`}
   </button>
 );
+
 const WrappedComponent = withNavigate(withLocation(withParams(MockComponent)));
 
-test('Provide Navigation to Component', () => {
-  const wrapper = mount(
-    <WrappedComponent />,
-  );
-  const btn = wrapper.find('#btn');
-  btn.simulate('click');
+test('Provides Navigation to Component', () => {
+  render(<WrappedComponent />);
+  const btn = screen.getByTestId('btn');
 
+  fireEvent.click(btn); // ✅ Replaces `simulate('click')`
```

Comment:
remove comment? 

[curious] How was this comment generated? Did you use any automated refactoring or AI tool?

Reply:
Potentially. Since I took over the PR from arbi-bom, it is possible some AI help was taken.